homeassistant/components/http/auth.py

```diff
@@ -40,10 +43,104 @@ def async_sign_path(hass, refresh_token_id, path, expiration):
 
 
 @callback
-def setup_auth(app, trusted_networks, api_password):
+def setup_auth(hass, app):
     """Create auth middleware for the app."""
     old_auth_warning = set()
 
+    support_legacy = hass.auth.support_legacy
+    if support_legacy:
+        _LOGGER.warning("legacy_api_password support has been enabled.")
+
+    trusted_networks = []
+    for prv in hass.auth.auth_providers:
+        if prv.type == 'trusted_networks':
+            trusted_networks += prv.trusted_networks
+
+    async def async_validate_auth_header(request):
```

Comment:
What's the benefit of pulling these into the `setup_auth` method ? I would prefer them stay outside

Reply:
Those two functions are not meant to be called by other program, it was pull out as a separate method only because the `auth_middleware` function became too long. 

Both functions also not immutable, they inserted user into request object, they are not purely utility functions. 

Also, I need access `hass` and `old_auth_warning` inside those method, pass in `hass` is acceptable, but pass in `old_auth_warning` looks weird.